shims/common/src/main/scala/org/apache/gluten/execution/datasource/GlutenFormatWriterInjects.scala

```diff
@@ -58,17 +56,42 @@ trait GlutenRowSplitter {
       reserve_partition_columns: Boolean = false): BlockStripes
 }
 
-object GlutenRowSplitter {
-  private var INSTANCE: GlutenRowSplitter = _
+object GlutenFormatFactory {
```

Comment:
nit: The file can be renamed to `GlutenFormatFactory.scala`

Reply:
I will rename in the follow up PRs.

Thanks!